Б-4.1

roman-net8 opened the pull request:
1. Подключил к проекту Entity Framework Core Postgresql
-  Сделал Docker-compose файл конфигурация для запуска Postgres.
-  Настроил appsettings для хранения строки подключения к бд.

2. Добавил конфигурации для всех доменных моделей. 
- Добавил под каждую модель отдельный класс с конфигурацией. 
- Включил логирование запросов в базу данных, которые делает ef core с помощью LoggerFactory.
-  Сделал все названия в базе данных были в snake_case стиле. 
- Создал миграции, применить их.

roman-net8 committed:
1. Подключил к проекту Entity Framework Core Postgresql
 Сделал Docker-compose файл конфигурация для запуска Postgres.
 Настроил appsettings для хранения строки подключения к бд.

2. Добавил конфигурации для всех доменных моделей.
    Добавил под каждую модель отдельный класс с конфигурацией.
    Включил логирование запросов в базу данных, которые делает ef core с помощью LoggerFactory.
    Сделал все названия в базе данных были в snake_case стиле.
    Создал миграции, применить их.

YADARAN1 commented on backend/src/PetFamily.Domain/Models/Volunteers/Volunteer.cs:
```diff
@@ -1,27 +1,26 @@
-        public void AddRequisite(Pet pet) => _ownedPets.Add(pet);
     }
+
+    public string FirstName { get; private set; } = default!;
```
из этих свойств(firstName, middleName, lastName) лучше сделать VO

YADARAN1 commented on backend/src/PetFamily.Domain/Shared/Constants.cs:
```diff
@@ -0,0 +1,44 @@
+
+    public static class SocialNetworkConst
+    {
+        public const int MAX_FIRST_NAME_TEXT_LENGTH = 32;
```
? Не соответствующее название констант + некоторые из них дублируются, сделай для некоторых свойств общие константы

YADARAN1 commented on backend/src/PetFamily.Domain/Shared/SocialNetwork.cs:
```diff
@@ -0,0 +1,9 @@
+
+public record SocialNetwork
+{
+    public string FirstName { get; private set; } = default!;
```
Почему в соц.сетях у тебя ФИО? Переделать

YADARAN1 commented on backend/src/PetFamily.Infrastructure/ApplicationDbContext.cs:
```diff
@@ -0,0 +1,31 @@
+    private const string DATABASE = "Database";
+
+    public DbSet<Volunteer> Volunteers => Set<Volunteer>();
+    public DbSet<Pet> Pets => Set<Pet>();
```
DbSet животных удалить, мы взаимодействуем с животным через волонтёра

YADARAN1 commented on backend/src/PetFamily.Infrastructure/ApplicationDbContext.cs:
```diff
@@ -0,0 +1,31 @@
+
+    public DbSet<Volunteer> Volunteers => Set<Volunteer>();
+    public DbSet<Pet> Pets => Set<Pet>();
+    public DbSet<PetPhoto> PetPhotos => Set<PetPhoto>();
```
Это тоже убрать

YADARAN1 commented on backend/src/PetFamily.Infrastructure/Configurations/VolunteerConfiguration.cs:
```diff
@@ -0,0 +1,87 @@
+
+        builder.OwnsOne(v => v.Details, vb =>
+        {
+            vb.ToJson();
```
В скобках указать названия jsonb столбца в snakecase стиле

roman-net8 committed:
Исправления по коментариям к PetFamily/pull/7

YADARAN1 commented on backend/src/PetFamily.Domain/Shared/PersonFullName.cs:
```diff
@@ -0,0 +1,7 @@
+public record PersonFullName
+{
+    public string FirstName { get; private set; } = default!;
+    public string LastName { get; private set; } = default!;
```
у нас может быть отчество, а может и не быть, cделай ещё 1 свойство

YADARAN1 commented on backend/src/PetFamily.Domain/Models/Volunteers/Volunteer.cs:
```diff
@@ -10,8 +10,7 @@ private Volunteer(VolunteerId id) : base(id)
 
-    public string FirstName { get; private set; } = default!;
-    public string LastName { get; private set; } = default!;
+    public PersonFullName FullName { get; private set; } = default!;
```
просто FullName

YADARAN1 commented on backend/src/PetFamily.Infrastructure/Configurations/PetConfiguration.cs:
```diff
@@ -68,7 +68,7 @@ public void Configure(EntityTypeBuilder<Pet> builder)
 
         builder.OwnsMany(p => p.Requisites)
-            .ToJson();
+            .ToJson("json_data");
```
requisites_list

YADARAN1 commented on backend/src/PetFamily.Infrastructure/Configurations/VolunteerConfiguration.cs:
```diff
@@ -18,28 +18,32 @@ public void Configure(EntityTypeBuilder<Volunteer> builder)
-        builder.Property(v => v.FirstName)
-             .IsRequired()
-             .HasMaxLength(Constants.VolunteerConst.MAX_FIRST_NAME_TEXT_LENGTH);
+        builder.OwnsOne(v => v.FullName, vb =>
```
Cконфигурируй как ComplexProperty

YADARAN1 commented on backend/src/PetFamily.Infrastructure/Configurations/VolunteerConfiguration.cs:
```diff
@@ -50,36 +54,32 @@ public void Configure(EntityTypeBuilder<Volunteer> builder)
         builder.OwnsOne(v => v.Details, vb =>
         {
-            vb.ToJson();
+            vb.ToJson("json_data");
```
details_list

roman-net8 committed:
Исправления к PetFamily/pull/7/commits/243e91318fab8310d5d62ba2009889fc30ddc175
Добавил MiddleName,
сконфигурировал как ComplexProperty,
переименован  json_data, FullName.
